### ISSUE TITLE
[sale_sourced_by_line] To be discontinued?

### ISSUE THREAD

**Issue by JordiBForgeFlow** [OCA/sale-workflow#529]
I have recently been quite embarrassed by a Tweet from Fabien questioning the purpose of this module https://twitter.com/fpodoo/status/911511134817193984.

And he's right. Technically you can create a Route, applicable on sales order lines, that will create a separate picking, to be sourced from a separate warehouse.

![image](https://user-images.githubusercontent.com/7683926/30777332-802bdff0-a0b8-11e7-8409-aa93e2f0467e.png)

So, in light of this evidence, what is the purpose of the module?

Looking at other depending modules such as https://github.com/OCA/sale-workflow/tree/8.0/sale_sourced_by_line_sale_transport_multi_address, I can see that it is in some cases it is important to store the sourcing warehouse.

Also, in a related conversation with cedric krier, https://twitter.com/cedrickrier/status/911254082735759361, we reflected on the importance of storing the ship from and ship to in sales order lines, due to the principle that taxation should follow the flow of physical goods. It is important to know the origin and destination countries of each item in a sales order, so as to properly drive taxes. If you ship part of a sales order from another country, your taxes may differ. But this would require to reconsider how the taxes are computed in sales orders and invoices.

I am cc'ing previous contributors @yvaucher @jgrandguillaume @guewen, that perhaps can provide some more light into this.

**Comment by yvaucher** [comment]
@jbeficent This is a dependency for the modules #oca/vertical-ngo/logistic_order (dependency on sale_sourced_by_line_sale_transport_multi_address) and #oca/vertical-ngo/logistic_requisition

In this context of NGO we transformed the sale order into a logistic order. On the logistic order it was required to be able to set by hand a specific warehouse. And we needed the traceability in a context of international logistics exchanges. I don't remind if it was for taxes or accountability to donors.


Now this module had been created in version 7.0 and was ported early to version 8.0. So it could be a remainder of the 7.0.

@jgrandguillaume @gurneyalex might remember more clearly for what it was designed that way and for which specific cases.

**Comment by dvdhinesh** [comment]
AFAICT we can do that by Pull rule but like @pedrobaeza mentioned on that tweet this module eases the complexity and it is pretty straightforward for the end user on the UI. :1st_place_medal: 

We are using this module for a company that let the user place order on their Magento site as well as employees on the backend. User/Employee can choose the preferred warehouse per line while placing the order. This comes handy for that type of requirements. Plus we are not maintaining accounts/tax in that system and it was purely designated for the logistic purpose.

OCA also have another [module](https://github.com/OCA/purchase-workflow/tree/9.0/purchase_location_by_line) that serves the similar purpose on the purchase side but instead of the warehouse, this module will specify the location.

**Comment by ThijsvOers** [comment]
Hi, 
I have read this thread on choosing the warehouse on order line level and i want to add some extra info that may be useful if you want to extend the functionality.

I have worked for a software company with a deep vertical solution for the retail business (furniture). We had companies that had different stores in different cities. All these stores were in one database, and used the same products. As the products could be available in the different stores (or in different warehouses), it was possible to see the stock level of all the stores when adding a order line item. 

Basically it meant that you could sell products from other locations. See the preview where you see a sales order where I am adding a product to a order line. While adding i have the option to see all stores/warehouse and the stock level of each store. When I add the order line the product is immediately assigned to the sales order.

![afbeelding](https://user-images.githubusercontent.com/30256071/30862483-7a055e72-a2ce-11e7-9a0b-869e839c49ae.png)

In this case routes are not the solution, main reason is that you directly sell from stock (and the MTO route is not applicable).

**Comment by emmanueldominguez** [comment]
So is it worth the work to migrate this module to 11 and then to 12?

**Comment by LoisRForgeFlow** [comment]
I would say is not, did you find a case in which standard Odoo does not suffice?

**Comment by scaldewi** [comment]
> I have recently been quite embarrassed by a Tweet from Fabien questioning the purpose of this module https://twitter.com/fpodoo/status/911511134817193984.
> 
> And he's right. Technically you can create a Route, applicable on sales order lines, that will create a separate picking, to be sourced from a separate warehouse.

Seems like there is no "Propagate, cancel and split" any more. I only see "cancel next move" instead with Odoo 13.0 and 14.0.
From playing with configuration, Sale order does not split when selecting Routes accordingly in sale order.

For me it looks like there is demand for this module (again).
